### PR TITLE
Update installation instructions for Spack, Charm++, SpECTRE

### PR DESCRIPTION
**Pull request by fmahebert**
* Slight cleanup of dependencies list
* Update Spack instructions...
  *  Point to Spack docs for OpenSSL directions
  *  Update instructions to set up user environment with modules from Spack
* Update Charm++ and SpECTRE instructions for clarity

### CODE REVIEW

File: docs/MainSite/Installation.md [hunk 1]
```diff
@@ -88,45 +92,61 @@ spack install jemalloc # or from your package manager
...
-* Apply the Charm++ patch for your version *after* building Charm++ by running
-`git apply SPECTRE_ROOT/support/Charm/vx.y.z.patch` in the Charm++ directory
-* `mkdir build && cd build`
+After the dependencies have been installed, Charm++ and SpECTRE can be compiled.
+Follow these steps:
+
+1.  Clone [Charm++](http://charm.cs.illinois.edu/software) into `CHARM_DIR`,
+    again a directory of your choice.
+    To use a stable version of Charm++, use `git tag` to see tagged releases,
+    and `git checkout vx.y.z` to switch to a recent release.
```
Comment (nilsdeppe) [inline docs/MainSite/Installation.md]: I would prefer we actually specify a required version and update this. Charm++ updates rather slowly and each update breaks lots of things...

File: docs/MainSite/Installation.md [hunk 2]
```diff
@@ -88,45 +92,61 @@ spack install jemalloc # or from your package manager
...
+    Follow the directions
+    [here](http://charm.cs.illinois.edu/manuals/html/charm++/A.html) to compile
+    Charm++. The correct target is `charm++` and, for a personal machine, the
+    correct target architecture is likely to be `mpi-linux-x86_64`
```
Comment (nilsdeppe) [inline docs/MainSite/Installation.md]: For 6.7.1 on personal machines it should be `multicore-...`

File: docs/MainSite/Installation.md [hunk 3]
```diff
@@ -88,45 +92,61 @@ spack install jemalloc # or from your package manager
 spack install libxsmm
```
Comment (nilsdeppe) [inline docs/MainSite/Installation.md]: Would you mind adding GSL to this list?

File: docs/MainSite/Installation.md [hunk 4]
```diff
@@ -4,37 +4,40 @@ See LICENSE.txt for details.
...
 * [Charm++](http://charm.cs.illinois.edu/) (must be compiled from source)
 * [Git](https://git-scm.com/)
-* BLAS
+* BLAS (e.g. [OpenBLAS](http://www.openblas.net))
 * [Blaze](https://bitbucket.org/blaze-lib/blaze/overview)
 * [Boost](http://www.boost.org/)
 * [Brigand](https://github.com/edouarda/brigand)
-* [Catch](https://github.com/philsquared/Catch) (v1.6.1 or older)
+* [Catch](https://github.com/philsquared/Catch) v1.6.1 or older
 * [HDF5](https://support.hdfgroup.org/HDF5/) (non-mpi version on macOS)
```
Comment (nilsdeppe) [inline docs/MainSite/Installation.md]: Would you mind adding GSL to this list?

File: docs/MainSite/Installation.md (hunk 2 above)
Comment (nilsdeppe) [reply docs/MainSite/Installation.md]: We should also note that we will be providing specific instructions for various HPC systems

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Comment by fmahebert** [comment]
I will squash some of these commits and update the PR

**Pull request merged**